### PR TITLE
PERF: RangeIndex.round returns RangeIndex when possible

### PR DESCRIPTION
**Pull request by mroeschke**

### CODE REVIEW

File: pandas/core/indexes/range.py [hunk 1]
```diff
@@ -1165,6 +1165,43 @@ def any(self, *args, **kwargs) -> bool:
...
+        >>> idx.round(decimals=-1)
+        RangeIndex(start=10, stop=30, step=10)
+        >>> idx = pd.RangeIndex(10, 15, 1)
+        >>> idx.round(decimals=-1)
+        Index([10, 10, 10, 10, 10], dtype='int64')
+        """
+        if decimals >= 0:
+            return self.copy()
+        elif all(
+            getattr(self, attr) % 10**-decimals == 0 for attr in ("start", "step")
```
Comment (datapythonista) [inline pandas/core/indexes/range.py:1198]: Personal preference, but I'd find more readable to simply use `self.start % 10 ... and self.stop % 10 ...`.

File: pandas/core/indexes/range.py [hunk 2]
```diff
@@ -1165,6 +1165,43 @@ def any(self, *args, **kwargs) -> bool:
...
+        ):
+            # e.g. Range(10, 30, 10).round(-1) doesn't need rounding
```
Comment (datapythonista) [inline pandas/core/indexes/range.py:1200]:
```suggestion
            # e.g. RangeIndex(10, 30, 10).round(-1) doesn't need rounding
```

File: pandas/core/indexes/range.py [hunk 3]
```diff
@@ -1165,6 +1165,43 @@ def any(self, *args, **kwargs) -> bool:
...
+    # return type "RangeIndex" in supertype "Index"
+    def round(self, decimals: int = 0) -> Self | Index:  # type: ignore[override]
+        """
+        Round each value in the Index to the given number of decimals.
+
+        Parameters
+        ----------
+        decimals : int, optional
+            Number of decimal places to round to. If decimals is negative,
+            it specifies the number of positions to the left of the decimal point.
```
Comment (datapythonista) [inline pandas/core/indexes/range.py:1178]: I know this is from the original docstring, but for me it's not obvious what this means by reading it, maybe adding an example or two here would be helpful.

**Review by datapythonista — APPROVED** [review]
Looks good to me, added few suggestions.

File: pandas/core/indexes/range.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/indexes/range.py]: Fair point. I'll change this to your suggestion

File: pandas/core/indexes/range.py (hunk 2 above)
Comment (mroeschke) [reply pandas/core/indexes/range.py]: Thanks!

File: pandas/core/indexes/range.py (hunk 3 above)
Comment (mroeschke) [reply pandas/core/indexes/range.py]: Sure thing. I added an example for the negative case.

**Pull request merged**